engine/src/cmd/unit_const_cache.cpp

```diff
@@ -2,3 +2,15 @@
 
 template<> UnitConstCache::cache_map UnitConstCache::unit_cache = UnitConstCache::cache_map();
 template<> WeaponMeshCache::cache_map WeaponMeshCache::unit_cache = WeaponMeshCache::cache_map();
+
+bool operator==(const StringIntKey& a, const StringIntKey& b)
+{
+    return a.fac == b.fac && a.key == b.key;
+}
+
+bool operator<(const StringIntKey& a, const StringIntKey& b)
+{
+    if (a.fac != b.fac)
+        return a.fac < b.fac;
+    return a.key < b.key;
+}
```

Comment:
git wants there to be an extra blank line at the end of the file. (Small quibble)